### ISSUE TITLE
Feature Request : Usage API Ratelimit details to self throttle

### ISSUE THREAD

**Issue by jmrothst** [digitalocean/terraform-provider-digitalocean#473]
Hi there,

Thank you for opening an issue. Please note that we try to keep the Terraform issue tracker reserved for bug reports and feature requests. For general usage questions, please see: https://www.terraform.io/community.html.

### Terraform Version
Terraform v0.12.29
+ provider.cloudinit v1.0.0
+ provider.digitalocean v1.18.0
+ provider.random v2.2.1

### Affected Resource(s)
All Digital Ocean objects

If this issue appears to affect multiple resources, it may be an issue with Terraform's core, so please mention this.

### Terraform Configuration Files
N/A

### Debug Output
N/A

### Panic Output
N/A

### Expected Behavior
Terraform should be able to use Digital Ocean API and not trip the per hour or per minute rate limits (https://developers.digitalocean.com/documentation/v2/#rate-limit)

### Actual Behavior
With small terraform projects, can trip the per minute rate limit while still having plenty of per hour rate limit remaining. API provides an indicator for what the current per hour rate limit is, as well as the remaining limit. While no remaining per minute value is provided, a reasonable guess can be determined. When Error 429 is returned, this can be used to adjust for API back off.

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. Run any API Calls at over the per minute rate limit and under the per hour rate limit. Any combination of Plan and Apply will do this for a small project.

### Important Factoids
Are there anything atypical about your accounts that we should know? For example: Running in EC2 Classic? Custom version of OpenStack? Tight ACLs?

Nothing known special

### References
Are there any other GitHub issues (open or closed) or Pull Requests that should be linked here? For example:

https://developers.digitalocean.com/documentation/v2/#rate-limit

**Comment by andrewsomething** [comment]
Thanks opening this issue. It is very related to https://github.com/digitalocean/terraform-provider-digitalocean/issues/235 and likely something we would tackle together.

**Comment by aqos156** [comment]
@andrewsomething any info on this feature?

**Comment by benyanke** [comment]
Any updates on this? It's rather unfortunate that DO releases a TF provider that doesn't even work for their own cloud.

**Comment by andrewsomething** [comment]
In version 2.28.0 of this provider we added experimental support for automatically retrying requests that fail with connection errors, 429, or 500-level response codes. With the recently released version 2.30.0 of this provider, we have enabled this behavior by default. In addition to retrying requests, it will also back off as needed for requests triggering the rate limit. The `DIGITALOCEAN_HTTP_RETRY_MAX` environment variable or the `http_retry_max` argument in the provider configuration can be used to configure it. Setting them to `0` will disable this behavior altogether.
